inspect_buildpack_test.go

```diff
@@ -0,0 +1,604 @@
+						}
+					})
+
+					it("succeeds", func() {
+						inspectOptions := pack.InspectBuildpackOptions{
+							BuildpackName: "some/buildpack",
+							Daemon:        useDaemon,
+						}
+						info, err := subject.InspectBuildpack(inspectOptions)
+						h.AssertNil(t, err)
+
+						h.AssertEq(t, info, expectedInfo)
+					})
+				})
+			}
+		})
+	})
+	when("failure cases", func() {
+		when("invalid buildpack name", func() {
+			it("returns an error", func() {
+				invalidBuildpackName := ""
+				inspectOptions := pack.InspectBuildpackOptions{
+					BuildpackName: invalidBuildpackName,
+				}
+				_, err := subject.InspectBuildpack(inspectOptions)
+
+				h.AssertError(t, err, "unable to handle locator ")
+				h.AssertTrue(t, !errors.Is(err, image.ErrNotFound))
+			})
+		})
+		when("buildpack image", func() {
+			when("unable to fetch buildpack image", func() {
+				it.Before(func() {
+					mockImageFetcher.EXPECT().Fetch(gomock.Any(), "missing/buildpack", true, config.PullNever).Return(nil, errors.Wrapf(image.ErrNotFound, "big bad error"))
+				})
+				it("returns an ErrNotFound error", func() {
+					inspectOptions := pack.InspectBuildpackOptions{
+						BuildpackName: "missing/buildpack",
+						Daemon:        true,
+					}
+					_, err := subject.InspectBuildpack(inspectOptions)
+					h.AssertTrue(t, errors.Is(err, image.ErrNotFound))
+				})
+			})
+
+			when("image does not have buildpackage metadata", func() {
+				it.Before(func() {
+					fakeImage := fakes.NewImage("empty", "", nil)
+					h.AssertNil(t, fakeImage.SetLabel(dist.BuildpackLayersLabel, ":::"))
+					mockImageFetcher.EXPECT().Fetch(gomock.Any(), "missing-metadata/buildpack", true, config.PullNever).Return(fakeImage, nil)
+				})
+				it("returns an error", func() {
+					inspectOptions := pack.InspectBuildpackOptions{
+						BuildpackName: "missing-metadata/buildpack",
+						Daemon:        true,
+					}
+					_, err := subject.InspectBuildpack(inspectOptions)
+
+					h.AssertError(t, err, fmt.Sprintf("unable to get image label %s", dist.BuildpackLayersLabel))
+					h.AssertFalse(t, errors.Is(err, image.ErrNotFound))
+				})
+			})
+		})
+		when("buildpack archive", func() {
+			when("archive is not a buildpack", func() {
+				it.Before(func() {
+					invalidBuildpackPath := filepath.Join(tmpDir, "fake-buildpack-path")
+					h.AssertNil(t, ioutil.WriteFile(invalidBuildpackPath, []byte("not a buildpack"), os.ModePerm))
+
+					mockDownloader.EXPECT().Download(gomock.Any(), "https://invalid/buildpack").Return(blob.NewBlob(invalidBuildpackPath), nil)
+				})
+				it("returns an error", func() {
+					inspectOptions := pack.InspectBuildpackOptions{
+						BuildpackName: "https://invalid/buildpack",
+						Daemon:        true,
+					}
+
+					_, err := subject.InspectBuildpack(inspectOptions)
+					h.AssertNotNil(t, err)
+					h.AssertTrue(t, !errors.Is(err, image.ErrNotFound))
```

Comment:
Ya these extra comparisons are just to be explicit about the error 'type'. The `image.ErrNotFound` is used to differentiate the following cases:

- `fetcher.Fetch(...)` call failed to retrieve an image because it does not exist (returns `image.ErrNotFound`, we just tried to fetch an image that does not exist & can probably continue program execution ).
- `fetcher.Fetch(...)` returned some other error when attempting to retrieve the image. (return any other error type, something terrible happened and we should probably exit)

Happy to add string matching as well. But I think we might want these explicit 'type' checks.